helix-term/src/grammars.rs

```diff
@@ -0,0 +1,290 @@
+use anyhow::{anyhow, Context, Result};
+use std::fs;
+use std::time::SystemTime;
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+    sync::mpsc::channel,
+};
+
+use helix_core::syntax::{GrammarConfiguration, GrammarSource, DYLIB_EXTENSION};
+
+const BUILD_TARGET: &str = env!("BUILD_TARGET");
+const REMOTE_NAME: &str = "helix-origin";
+
+pub fn fetch_grammars() {
+    run_parallel(get_grammar_configs(), fetch_grammar);
+}
+
+pub fn build_grammars() {
+    run_parallel(get_grammar_configs(), build_grammar);
+}
+
+fn run_parallel<F>(grammars: Vec<GrammarConfiguration>, job: F)
+where
+    F: Fn(GrammarConfiguration) + std::marker::Send + 'static + Copy,
+{
+    let mut n_jobs = 0;
+    let pool = threadpool::Builder::new().build();
+    let (tx, rx) = channel();
+
+    for grammar in grammars {
+        let tx = tx.clone();
+        n_jobs += 1;
+
+        pool.execute(move || {
+            job(grammar);
+
+            // report progress
+            tx.send(1).unwrap();
+        });
+    }
+    pool.join();
+
+    assert_eq!(rx.try_iter().sum::<usize>(), n_jobs);
```

Comment:
Should we print a more useful error message? If a grammar fails to fetch or clone, this will be very cryptic.